### PR TITLE
Add unit tests for pipeline_helper.py (part 2)

### PR DESCRIPTION
**Pull request by haozhestat**
In this pull request, we add unit tests for three functions in the  `pipeline_helper.py` file:
- [x] _apply_linux_runsettings
- [x] apply_recommended_runsettings

### CODE REVIEW

File: tests/tests_pipeline/test_pipeline_helper.py [hunk 1]
```diff
@@ -228,3 +228,103 @@ def test_apply_hdi_runsettings(capsys):
...
+        process_count_per_node=3 if mpi else None,
+    )
+
+    out, _ = capsys.readouterr()
+    sys.stdout.write(out)
+
+    # Testing mpi runsetting parameter configuration
+    if mpi:
+        assert module_instance.runsettings.resource_layout.node_count == 2
+        assert module_instance.runsettings.resource_layout.process_count_per_node is 3
```
Comment (fuhuifang) [inline tests/tests_pipeline/test_pipeline_helper.py:258]: == 3?

File: tests/tests_pipeline/test_pipeline_helper.py [hunk 2]
```diff
@@ -228,3 +228,103 @@ def test_apply_hdi_runsettings(capsys):
...
+
+    # Testing compute target configuration
+    if gpu:
+        assert module_instance.runsettings.target == "gpu-cluster"
+        assert f"Using target gpu-cluster for registered GPU module {module_name} from pipeline classAMLPipelineHelper"
```
Comment (fuhuifang) [inline tests/tests_pipeline/test_pipeline_helper.py:263]: I'm not familiar with this, but do you need to `assert XXX in out`?

File: tests/tests_pipeline/test_pipeline_helper.py [hunk 3]
```diff
@@ -228,3 +228,103 @@ def test_apply_hdi_runsettings(capsys):
...
+    pipeline_helper.apply_recommended_runsettings(
+        module_name=module_name,
+        module_instance=module_instance,
+    )
+
+    out, _ = capsys.readouterr()
+    sys.stdout.write(out)
+
+    if module_name == "convert2ss":
+        assert f"Module {module_name} detected as SCOPE: True" in out
```
[the remark this answers is not in the document]
Comment (haozhestat) [reply tests/tests_pipeline/test_pipeline_helper.py:321]: @fuhuifang: I think I had already added "@pytest.mark.parametrize". Could you double check?

File: tests/tests_pipeline/test_pipeline_helper.py [hunk 4]
```diff
@@ -228,3 +228,109 @@ def test_apply_hdi_runsettings(capsys):
...
+    else:
+        assert module_instance.runsettings.target == "cpu-cluster"
+        assert f"Using target cpu-cluster for registered CPU module {module_name} from pipeline classAMLPipelineHelper"
+
+    # Testing input and output mode
+    assert module_instance.inputs.input_path.mode == "download"
+    assert module_instance.outputs.output_path.output_mode == "upload"
+
+
+@pytest.mark.parametrize(
```
Comment (haozhestat) [inline tests/tests_pipeline/test_pipeline_helper.py:273]: @fuhuifang : This is where I added "@pytest.mark.parametrize".

Comment (fuhuifang) [reply tests/tests_pipeline/test_pipeline_helper.py:273]: @haozhestat Could you move the expected results into `@pytest.mark.parametrize`? https://github.com/Azure/shrike/blob/0d7edbbbf5eb66fd243b5908d247ae4e793c24ca/tests/tests_build/commands/test_prepare.py#L400

Comment (fuhuifang) [reply tests/tests_pipeline/test_pipeline_helper.py:273]: for apply_recommended_runsettings()

File: tests/tests_pipeline/test_pipeline_helper.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (fuhuifang) [reply tests/tests_pipeline/test_pipeline_helper.py:321]: something like (wrong syntax)
```
@pytest.mark.parametrize(
"module_name, expected_res", [["convert2ss", f"Module {module_name} detected as SCOPE: True" ], ["data_transfer", f"Module {module_name} detected as DATATRANSFER: True"]
)
...
assert expected_res in out
```

File: tests/tests_pipeline/test_pipeline_helper.py (hunk 1 above)
Comment (haozhestat) [reply tests/tests_pipeline/test_pipeline_helper.py]: Good catch. Addressed.

File: tests/tests_pipeline/test_pipeline_helper.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (haozhestat) [reply tests/tests_pipeline/test_pipeline_helper.py]: Got it. Addressed.

File: tests/tests_pipeline/test_pipeline_helper.py (hunk 2 above)
Comment (haozhestat) [reply tests/tests_pipeline/test_pipeline_helper.py]: Good catch. Addressed.

**Review by fuhuifang — APPROVED** [review]

**Pull request merged**